Library/Homebrew/keg_relocate.rb

```diff
@@ -70,14 +88,14 @@ def relocate_dynamic_linkage(_relocation)
 
   def prepare_relocation_to_placeholders
     relocation = Relocation.new
-    relocation.add_replacement_pair(:prefix, HOMEBREW_PREFIX.to_s, PREFIX_PLACEHOLDER)
-    relocation.add_replacement_pair(:cellar, HOMEBREW_CELLAR.to_s, CELLAR_PLACEHOLDER)
+    relocation.add_replacement_pair(:prefix, HOMEBREW_PREFIX.to_s, PREFIX_PLACEHOLDER, path: true)
+    relocation.add_replacement_pair(:cellar, HOMEBREW_CELLAR.to_s, CELLAR_PLACEHOLDER, path: true)
     # when HOMEBREW_PREFIX == HOMEBREW_REPOSITORY we should use HOMEBREW_PREFIX for all relocations to avoid
     # being unable to differentiate between them.
     if HOMEBREW_PREFIX != HOMEBREW_REPOSITORY
-      relocation.add_replacement_pair(:repository, HOMEBREW_REPOSITORY.to_s, REPOSITORY_PLACEHOLDER)
+      relocation.add_replacement_pair(:repository, HOMEBREW_REPOSITORY.to_s, REPOSITORY_PLACEHOLDER, path: true)
     end
-    relocation.add_replacement_pair(:library, HOMEBREW_LIBRARY.to_s, LIBRARY_PLACEHOLDER)
+    relocation.add_replacement_pair(:library, HOMEBREW_LIBRARY.to_s, LIBRARY_PLACEHOLDER, path: true)
     relocation.add_replacement_pair(:perl,
                                     %r{\A#!(?:/usr/bin/perl\d\.\d+|#{HOMEBREW_PREFIX}/opt/perl/bin/perl)( |$)}o,
                                     "#!#{PERL_PLACEHOLDER}\\1")
```

Comment:
I didn't add `path: true` here because it looks like it's matching shebangs not paths (cc @Bo98)

Reply:
That is correct.

I wonder if `path` could be inferred if the first parameter is a `Pathname`. I'm probably overcomplicating it though.

Reply:
A tricky thing is that there are some times (e.g. just above this) where a string is passed but should still be treated as a pathname. So, yeah, I think that could be done but we probably need the `path` argument anyway.

I'm leaning toward leaving as is unless you think it makes more sense to automatically detect it.